src/App.tsx

```diff
@@ -1,17 +1,45 @@
 import '@redhat-cloud-services/frontend-components-utilities/styles/_all';
 import 'react18-json-view/src/style.css';
 import NotificationsPortal from '@redhat-cloud-services/frontend-components-notifications/NotificationPortal';
+import { Bullseye, Spinner } from '@patternfly/react-core';
+import { useEffect, /* useMemo,*/ useState } from 'react';
+// import { useLocation } from 'react-router-dom';
+// import { last } from 'lodash';
+
 import Routes from './Routes';
 import { useAppContext } from './middleware/AppContext';
 import { NoPermissionsPage } from './components/NoPermissionsPage/NoPermissionsPage';
-
-import { Bullseye, Spinner } from '@patternfly/react-core';
+import { ContentOrigin, FilterData } from './services/Content/ContentApi';
+import { useContentListQuery } from './services/Content/ContentQueries';
+import { perPageKey } from './Pages/ContentListTable/ContentListTable';
+// import { REPOSITORIES_ROUTE } from './Routes/constants';
 
 export default function App() {
-  const { rbac, isFetchingFeatures } = useAppContext();
+  const { rbac, isFetchingFeatures, zeroState, setZeroState } = useAppContext();
+  const storedPerPage = Number(localStorage.getItem(perPageKey)) || 20;
+  // const { pathname } = useLocation();
+  // const isDefaultRoute = useMemo(() => last(pathname.split('/')) === REPOSITORIES_ROUTE, [pathname]);
```

Comment:
@adewar once i rebased some of this logic broke (the zero state page would show up even if there were repositories). it works now that i've commented it out, but i know we added this to make sure the useContentListQuery wasn't being called when it didn't need to be. let me know if you have any ideas on how to add this functionality back in